app/javascript/articles/FeaturedArticle.jsx

```diff
@@ -0,0 +1,107 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import {
+  TagList,
+  SaveButton,
+  ReadingTime,
+  ReactionsCount,
+  CommentsCount,
+} from './components';
+import { articlePropTypes } from '../src/components/common-prop-types';
+
+// TODO: cloud_cover_url
+const cloud_cover_url = url => url; // eslint-disable-line camelcase
+
+export const FeaturedArticle = ({
+  article,
+  isBookmarked,
+  reactionsIcon,
+  commentsIcon,
+}) => {
+  return (
+    <div>
+      <div
+        id="featured-story-marker"
+        data-featured-article={`articles-${article.id}`}
+      />
+      <img
+        src={cloud_cover_url(article.main_image)}
+        style={{ display: 'none' }}
+        alt={article.title}
+      />
+      <a
+        href={article.path}
+        id={`article-link-${article.id}`}
+        className="index-article-link"
+        aria-label="Main Story"
+        data-featured-article={`articles-${article.id}`}
+      >
+        <div
+          className="single-article big-article"
+          data-content-user-id={article.user_id}
+        >
+          <div
+            className="picture image-final"
+            style={{
+              backgroundColor: article.main_image_background_hex_color,
+              backgroundImage: `url(${cloud_cover_url(article.main_image)})`,
```

Comment:
This is the `cloud_cover_url` function I was talking about @benhalpern. I'm faking it here on the client-side by simply returning the `main_image` URL, but I imagine the `cloud_cover_url` server-side function returns a CDN URL?

Reply:
Ah yes. @joshpuetz I think `main_image` should deliver the fully CDN-ified version, e.g. `cloud_cover_url` which had been working via view helper...

Similar to `cached_user`/`user` we can let the server worry about sending the appropriate version.

Reply:
We would need to return the cloud URL in the feed for the first story, i.e. featured story. I can't do it at build time like the other ERB/webpacker stuff, because the feature story changes.

Reply:
Ooops, responded on stale comments. 🙃

Reply:
So I'll assume the proper URL is being sent for the main image then.

Reply:
Just checked the feed endpoint, and we're sending what I think are non-cdn-ified locations for images (ex: https://dev-to-uploads.s3.amazonaws.com/i/9bmcx1wglljih84nx65u.png). Will adjust in the endpoint @nickytonline!